tracer/src/Datadog.Trace/Configuration/ImmutableAzureAppServiceSettings.cs

```diff
@@ -183,5 +183,15 @@ private static bool ShouldSkipClientSpanWithinFunctions(Scope? scope)
 
             return null;
         }
+
+        // Checks for azure functions specific env vars. Used when configuration hasn't been initialized yet.
+        internal static bool GetIsAzureConsumptionPlanFunction()
+        {
+            var isFunction = Environment.GetEnvironmentVariable(ConfigurationKeys.AzureAppService.FunctionsExtensionVersionKey) != null && Environment.GetEnvironmentVariable(ConfigurationKeys.AzureAppService.FunctionsWorkerRuntimeKey) != null;
+
+            var websiteSKU = Environment.GetEnvironmentVariable(ConfigurationKeys.AzureAppService.WebsiteSKU);
```

Comment:
windows consumption plan functions have `WEBSITE_SKU=dynamic`, while linux consumption plan functions omit the env var.

Reply:
AAS plan functions should always have a SKU set, so there shouldn't be any risk the mini agent is started in non consumption plan environments.